Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -1629,13 +1618,16 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
             namespace: namespace,
             document: docForStorage,
             writePending: true)
-        let config = try nsConfig.sync(id: result.insertedId!)
-        try config.setSomePendingWrites(atTime: logicalT, changeEvent: event)
 
-        deferredEventEmittingBlock = {
+        try nsConfig.nsLock.write {
```

Comment:
[minor q] why is there a gap in what is locked here. Everything before the trigger listening call should fall under the same critical section.